pkgs/top-level/all-packages.nix

```diff
@@ -9095,11 +9095,7 @@ in
   bison = bison3;
   yacc = bison; # TODO: move to aliases.nix
 
-  blackmagic = callPackage ../development/tools/misc/blackmagic {
-    stdenv = gcc6Stdenv;
-    gcc-arm-embedded = pkgsCross.arm-embedded.buildPackages.gcc;
-    binutils-arm-embedded = pkgsCross.arm-embedded.buildPackages.binutils;
```

Comment:
This was added in https://github.com/NixOS/nixpkgs/pull/48286 and judging from that change I think we should preserve this, though we don't need `binutils-arm-embedded`?

Reply:
Unfortunately, using `pkgsCross.arm-embedded.buildPackages.gcc` results in a bunch of linker errors:

```
  LD      blackmagic
/nix/store/agqr6rs1qzmgfxdyb7vgxikpnf2c8fkm-arm-none-eabi-binutils-2.31.1/bin/arm-none-eabi-ld: error: blackmagic uses VFP register arguments, /nix/store/i6zr1rlhsi60yhawrl08iffiv9yjvj26-newlib-3.1.0-arm-none-eabi/arm-none-eabi/lib/libc.a(lib_a-abort.o) does not
/nix/store/agqr6rs1qzmgfxdyb7vgxikpnf2c8fkm-arm-none-eabi-binutils-2.31.1/bin/arm-none-eabi-ld: failed to merge target specific data of file /nix/store/i6zr1rlhsi60yhawrl08iffiv9yjvj26-newlib-3.1.0-arm-none-eabi/arm-none-eabi/lib/libc.a(lib_a-abort.o)
/nix/store/agqr6rs1qzmgfxdyb7vgxikpnf2c8fkm-arm-none-eabi-binutils-2.31.1/bin/arm-none-eabi-ld: error: blackmagic uses VFP register arguments, /nix/store/i6zr1rlhsi60yhawrl08iffiv9yjvj26-newlib-3.1.0-arm-none-eabi/arm-none-eabi/lib/libc.a(lib_a-atoi.o) does not
/nix/store/agqr6rs1qzmgfxdyb7vgxikpnf2c8fkm-arm-none-eabi-binutils-2.31.1/bin/arm-none-eabi-ld: failed to merge target specific data of file /nix/store/i6zr1rlhsi60yhawrl08iffiv9yjvj26-newlib-3.1.0-arm-none-eabi/arm-none-eabi/lib/libc.a(lib_a-atoi.o)
[...]
```

I'm not sure how to resolve this; I think it might be a limitation of nixpkgs' cross-compilation not targetting the right ARM cores?

Reply:
https://github.com/NixOS/nixpkgs/issues/51907 seems related and suggests that it might be best to stick with `gcc-arm-embedded` here for targetting bare metal.

Reply:
Yep I think that's correct @emilazy.